lib/app/features/home/match_details/match_details_tabs/match_summary_tab/elements/first_half_events_display.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:flutter/material.dart';
+import 'package:score_live/app/custom_widgets/event_text_template.dart';
+import 'package:score_live/models/live_match_model/live_match_model.dart';
+import 'package:score_live/models/match_events_model/match_events_model.dart';
+import 'package:score_live/presentation/constants/app_colors.dart';
+import 'package:score_live/presentation/constants/app_const_variables.dart';
+
+class FirstHalfEventsDisplay extends StatelessWidget {
+  const FirstHalfEventsDisplay({super.key, required this.liveMatch, required this.matchEvents});
+
+  final LiveMatchModel liveMatch;
+  final List<MatchEventsModel> matchEvents;
+
+  @override
+  Widget build(BuildContext context) {
+    final int homeTeamID = liveMatch.teams?.home?.id ?? AppConstVariables.intPlaceholder;
+    final int awayTeamID = liveMatch.teams?.away?.id ?? AppConstVariables.intPlaceholder;
+    return Container(
+      decoration: BoxDecoration(
+        borderRadius: BorderRadius.circular(12),
+        color: AppColors.listTileGrey,
+      ),
+      child: ListView(
+        shrinkWrap: true,
+        primary: false,
+        children: [
+          Row(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              Expanded(
+                child: Column(
+                  children: [
+                    for (final event in matchEvents) ...[
+                      if (event.team?.id != null && event.time?.elapsed != null && event.player?.name != null)
```

Comment:
I would write an extension to event list to get only the valid ones -> so filter them by `event.team?.id != null && event.time?.elapsed != null && event.player?.name != null` - and use here as `matchEvents.validEvents` or something similar.

Reply:
right, done